pkg/gitprovider/azure/azure.go

```diff
@@ -181,6 +181,83 @@ func (p *provider) ListPullRequests(
 	return pts, nil
 }
 
+// MergePullRequest implements gitprovider.Interface.
+func (p *provider) MergePullRequest(
+	ctx context.Context,
+	id int64,
+) (*gitprovider.PullRequest, bool, error) {
+	var pr *gitprovider.PullRequest
+
+	gitClient, err := adogit.NewClient(ctx, p.connection)
+	if err != nil {
+		return nil, false, fmt.Errorf("error creating Azure DevOps client: %w", err)
+	}
+
+	// Get the current PR to check its status and get the last merge source commit
+	adoPR, err := gitClient.GetPullRequest(ctx, adogit.GetPullRequestArgs{
+		Project:       &p.project,
+		RepositoryId:  &p.repo,
+		PullRequestId: ptr.To(int(id)),
+	})
+	if err != nil {
+		return nil, false, fmt.Errorf("error getting pull request %d: %w", id, err)
+	}
+	if adoPR == nil {
+		return nil, false, fmt.Errorf("pull request %d not found", id)
+	}
+
+	status := ptr.Deref(adoPR.Status, adogit.PullRequestStatusValues.NotSet)
+	mergeStatus := ptr.Deref(adoPR.MergeStatus, adogit.PullRequestAsyncStatusValues.NotSet)
+
+	switch status {
+	case adogit.PullRequestStatusValues.Completed:
+		pr, err = convertADOPullRequest(adoPR)
+		if err != nil {
+			return nil, false, fmt.Errorf("error converting pull request %d: %w", id, err)
+		}
+		return pr, true, nil
+	case adogit.PullRequestStatusValues.Abandoned:
+		return nil, false, fmt.Errorf("pull request %d is abandoned", id)
+	case adogit.PullRequestStatusValues.Active:
+		if ptr.Deref(adoPR.IsDraft, false) {
+			// Draft PRs should not be merged
+			return nil, false, nil
```

Comment:
For the benefit of future readers of this code:

```suggestion
		// Draft PRs can have a merge status of `succeeded`, but aren't actually
		// mergable, so we explicitly check for draft status.
		if ptr.Deref(adoPR.IsDraft, false) {
			return nil, false, nil
```